app/services/onetime/index_products_with_allowed_offer_codes.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class Onetime::IndexProductsWithAllowedOfferCodes
```

Comment:
@EmCousin noticed production already has some `BLACKFRIDAY2025` offer codes, we need to run this script after deploying this PR.